src/chunk.c

```diff
@@ -3261,6 +3261,28 @@ ts_chunk_get_chunk_ids_by_hypertable_id(int32 hypertable_id)
 	return chunkids;
 }
 
+List *
+ts_chunk_get_all_chunk_ids(LOCKMODE lockmode)
+{
+	List *chunkids = NIL;
+	ScanIterator iterator = ts_scan_iterator_create(CHUNK, lockmode, CurrentMemoryContext);
+	ts_scan_iterator_set_index(&iterator, CHUNK, CHUNK_ID_INDEX);
+	ts_scan_iterator_scan_key_init(&iterator,
+								   Anum_chunk_idx_id,
+								   BTEqualStrategyNumber,
+								   F_INT4GE,
+								   Int32GetDatum(0));
+	ts_scanner_foreach(&iterator)
```

Comment:
Would be nice if the join could be done in this loop using a per-tuple memory context to avoid memory O(N) memory usage.